ligo/test/Test/Ligo/BaseDAO/Management.hs

```diff
@@ -0,0 +1,269 @@
+-- SPDX-FileCopyrightText: 2020 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+--
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+-- For all the incomplete list pattern matches in the calls to with
+-- withOriginated func
+module Test.Ligo.BaseDAO.Management
+  ( test_BaseDAO_Management
+  ) where
+
+import Universum
+
+import Test.Tasty (TestTree, testGroup)
+
+import Lorentz as L
+import Michelson.Typed (convertContract)
+import Michelson.Typed.Convert (untypeValue)
+import Michelson.Untyped.Entrypoints (unsafeBuildEpName)
+import Morley.Nettest
+import Morley.Nettest.Tasty (nettestScenarioCaps)
+import Tezos.Core (unsafeMkMutez)
+
+import Ligo.BaseDAO.Contract
+import Util.Named
+import Ligo.BaseDAO.Types
+
+-- | Function that originates the contract and also make a bunch of
+-- address (the `addrCount` arg determines the count) for use within
+-- the tests. It is not pretty, but IMO it makes the test a bit less
+-- verbose.
+withOriginated
+  :: MonadNettest caps base m
+  => Integer
+  -> ([Address] -> FullStorage)
+  -> ([Address] -> TAddress Parameter -> m a)
+  -> m a
+withOriginated addrCount storageFn tests = do
+  addresses <- mapM (\x -> newAddress $ "address" <> (show x)) [1 ..addrCount]
+  baseDao <- originateUntyped $ UntypedOriginateData
+    { uodFrom = nettestAddress
+    , uodName = "BaseDAO Test Contract"
+    , uodBalance = zeroMutez
+    , uodStorage = untypeValue $ toVal $ storageFn addresses
+    , uodContract = convertContract baseDAOContractLigo
+    }
+  tests addresses (TAddress baseDao)
```

Comment:
I was wondering if it is possible to simply share the tests with lorentz basedao. I've done this with my tests, but since the tests are done differently here so I'm not sure. Example share tests: https://github.com/tqtezos/baseDAO/blob/master/src/BaseDAO/ShareTest/FA2.hs

The idea is to use the same tests but with different originate function. The errors for now are not the same (errors in ligo version are all string), but it will be the same after I finished this #106 .

Reply:
@rinn7e Yea, it should be possible (if the errors are changed after #106), and I did try to do it a little, but when I looked for a constraint like `ParameterC`, I missed the one in `Lorentz.Contract.BaseDAO.Types`. I was also not sure if made sense to reuse it, as the issue asked to reimplement tests due to difference in Haskell and Ligo version. Anyway I didn't look too closely, I guess we can revisit it after #106. 

@pasqu4le Does it make sense to open an issue?

Reply:
Yeah that's my bad, when I wrote that issue it just happened that it was clear we couldn't simply convert between the LIGO and Lorentz version of the contract and reuse all the test groups that we already had.

However Moremi has since made the test re-use as many scenarios as possible successfully, even tho it had to introduce differences in handling the failures.

Since we already have the new tests (will check them now BTW) I think it makes sense to merge this MR already and revisit them in a follow-up issue after #106 is merged as well. I'll make sure to open it.